Cargo.toml

```diff
@@ -22,4 +22,5 @@ edition = "2018"
 repository = "https://github.com/not-fl3/nanoserde"
 
 [dependencies]
+hashbrown = "0.12.3"
```

Comment:
Maybe make "hashbrown" optional?  

I am not aware of `no_std` best practices, but I would prefer "nanoserde" to have some sort of a cargo feature replacing std::HashMap with `hashbrown`  and therefore making it "no_std" compatible

To make switching to `hashbrown` optional for use cases where depending on std is ok